service/application/src/main/java/com/solace/maas/ep/event/management/agent/cli/EmaCommandLine.java

```diff
@@ -0,0 +1,209 @@
+import com.solace.maas.ep.event.management.agent.service.ImportService;
+import com.solace.maas.ep.event.management.agent.service.MessagingServiceDelegateServiceImpl;
+import com.solace.maas.ep.event.management.agent.service.ScanStatusService;
+import com.solace.maas.ep.event.management.agent.util.IDGenerator;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@Component
+@Order(1)
+@Slf4j
+public class EmaCommandLine implements CommandLineRunner {
+    private final ScanManager scanManager;
+    private final ImportService importService;
+    private final ScanStatusService scanStatusService;
+    private final IDGenerator idGenerator;
+    private final DataCollectionFileService dataCollectionFileService;
+    private final MessagingServiceDelegateServiceImpl messagingServiceDelegateService;
+
+    public EmaCommandLine(ScanManager scanManager,
+                          ImportService importService,
+                          ScanStatusService scanStatusService,
+                          IDGenerator idGenerator,
+                          DataCollectionFileService dataCollectionFileService,
+                          MessagingServiceDelegateServiceImpl messagingServiceDelegateService) {
+        this.scanManager = scanManager;
+        this.importService = importService;
+        this.scanStatusService = scanStatusService;
+        this.idGenerator = idGenerator;
+        this.dataCollectionFileService = dataCollectionFileService;
+        this.messagingServiceDelegateService = messagingServiceDelegateService;
+    }
+
+    @Override
+    public void run(String... args) throws Exception {
+        if (args.length > 2) {
+
+            String type = args[0];
+
+            if ("scan".equals(type)) {
+                String messagingServiceId = args[1];
+                String filePathAndName = args[2];
+
+                runScan(messagingServiceId, filePathAndName);
+            } else {
+                log.error("Unknown command: {}", type);
+            }
+        } else {
+            log.error("Not enough arguments passed to the application.");
+        }
+    }
+
+    private void runScan(String messagingServiceId, String filePathAndName) throws InterruptedException, IOException {
+        MessagingServiceEntity messagingServiceEntity = messagingServiceDelegateService.getMessagingServiceById(messagingServiceId);
+
+        ScanRequestBO scanRequestBO = new ScanRequestBO();
+        scanRequestBO.setMessagingServiceId(messagingServiceId);
+        scanRequestBO.setScanId(idGenerator.generateRandomUniqueId());
```

Comment:
We can introduce [Builder](https://projectlombok.org/features/Builder) to the BO objects which makes code easier to write and read.

Reply:
Added builder pattern for constructing the scan request.